lib/jarvis/workpool.rb

```diff
@@ -0,0 +1,52 @@
+require "concurrent"
+
+class WorkPool
+  ADMINISTRATIVE = :ADMINISTRATIVE
+  NORMAL = :NORMAL
+
+  class InvalidWorkPoolName < StandardError; end
+
+  class << self
+    def setup_singleton
+      @singleton = self.new
+      nil
+    end
+
+    def fetch(name)
+      @singleton.fetch(name)
+    end
+
+    def post(name, &block)
+      fetch(name).post(&block)
+    end
+  end
```

Comment:
(For posterity)

Singleton disables Class.new, and I want individual instances (if the programmer desires) in order to facilitate testing. Long term, I will remove this singleton entirely.